### DISCUSSION TITLE
Proposal for supporting semantic highlighting

### DISCUSSION THREAD

**Discussion by qobilidop** [p4lang/p4analyzer#23] [Proposals]
Semantic highlighting is a feature supported by LSP[1], and major editors/IDEs (like VS Code[2]) through LSP. The benefits of semantic highlighting are explained in more detail in [2]. I think it is a very useful feature, and a natural fit for p4analyzer to implement. As a reference, rust-analyzer has implemented this feature already[3].

Reference:
- [1] https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_semanticTokens
- [2] https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide
- [3] https://github.com/rust-lang/rust-analyzer/issues/604
- [4] https://github.com/build2/build2/issues/109

**Comment by timjroberts** [comment]
Yeah we do intend to look at using Semantic Tokens once we've got some good ground on the core. I'll open a Proposal Issue for this.
